extensions/kotlin/build.gradle.kts

```
@@ -48,6 +48,8 @@ tasks {
   withType(KotlinCompile::class) {
     kotlinOptions {
       jvmTarget = "1.8"
+      // earliest version supported by kotlin 1.9
+      languageVersion = "1.4"
```

#5983 defines our minimum kotlin language version to be 1.6. 
```suggestion
      languageVersion = "1.6"
```